screens/CheckoutScreen.js

```diff
@@ -33,98 +32,105 @@ export default class CheckoutScreen extends React.Component {
+          (prevState.totalBalance + priceDifference) % rewardDollarValue < 0
+            ? rewardDollarValue + ((prevState.totalBalance + priceDifference) % rewardDollarValue)
+            : (prevState.totalBalance + priceDifference) % rewardDollarValue,
+        rewardsApplied: prevState.rewardsApplied - rewardsToUndo
+      }));
+    }
   };
 
-  // Sets total points earned from transaction in state.
-  setTotalPoints = () => {
-    let points = 0;
-    // Iterate over lineItems in cart
-    Object.values(this.state.cart).forEach(lineItem => {
-      points += lineItem.points * lineItem.quantity;
-    });
-    points -= this.state.rewardsApplied * 500;
-    if (points < 0) {
-      console.error('Total points less than 0!');
+  // Calculates total points earned from transaction
+  // Accounts for lineItem individual point values and not allowing points to be earned with rewards daollrs
+  getPointsEarned = () => {
+    let pointsEarned = Object.values(this.state.cart).reduce(
+      (points, lineItem) => points + lineItem.points * lineItem.quantity,
+      0
+    );
+    // Customer cannot earn points with rewards dollars; assumes a reward's point multiplier per dollar is 100 pts
+    pointsEarned -= this.state.rewardsApplied * rewardDollarValue * 100;
+
+    // TODO this might be a design edge case now that rewards applied can bring the technical balance to a negative
+    if (pointsEarned < 0) {
+      if (this.state.totalBalance > 0) {
+        console.log(
+          'Total points less than 0! This is likely a bug, unless the value of various items is < 100 pts per item, since the real balance is positive. '
+        );
+      }
+      // Otherwise, expected - value of rewards applied > value of items in cart
+      pointsEarned = 0;
     }
-    this.setState({ totalPoints: points });
-    return points;
+    return pointsEarned;
   };
 
   // Handles submit when clerk selects "CHECKOUT".
   handleSubmit = () => {
-    const totalPoints = this.setTotalPoints();
-    this.displayConfirmation(totalPoints);
-  };
+    const { rewardsApplied, totalBalance } = this.state;
 
-  // Generates the confirmation message based on items in cart, points earned,
-  // and total spent.
-  generateConfirmationMessage = totalPoints => {
-    let msg = 'Transaction Items:\n\n';
-    // Iterate over lineItems in cart
-    Object.values(this.state.cart).forEach(lineItem => {
-      if (lineItem.quantity > 0) {
-        msg = msg.concat(`${lineItem.quantity} x ${lineItem.name}\n`);
-      }
-    });
-    msg = msg.concat(`\nRewards Redeemed: ${this.state.rewardsApplied}\n`);
-    // Adding total price and total points earned to message. Must be called after setTotalPoints()
-    // in handleSubmit() for updated amount.
-    msg = msg.concat(`\nTotal Price: $${this.state.totalPrice.toFixed(2)}\n`);
-    msg = msg.concat(`Total Points Earned: ${totalPoints}`);
-    return msg;
-  };
+    // Calculate actual discount and sale; handle negative balances
```

Comment:
@thumn This is where subtotal gets calculated!

Reply:
ty!